src/util/ecdsa.rs

```diff
@@ -453,11 +454,33 @@ impl EcdsaSig {
     }
 }
 
+impl fmt::Display for EcdsaSig {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}:{}", self.sig.serialize_der().to_hex(), self.hash_ty as u8)
```

Comment:
In 70515b322bd9c90e0d8133e6c9c3fe08986f0092:

Can you use `hex::format_hex` here to avoid the allocation?

Reply:
Fixed this too